### PR TITLE
Feature/15800

### PR DESCRIPTION
**Pull request by nimble-dfe**
**What is the change?**

Added model binding for NTI Under consideration error messages
**Why do we need the change?**

Error message changes and model binding

**What is the impact?**

Change to page error handling
**Azure DevOps Ticket**
125800

### CODE REVIEW

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Edit.cshtml.cs [hunk 1]
```diff
@@ -69,11 +71,28 @@ public async Task<IActionResult> OnPostAsync()
...
 			try
 			{
+				
+				if (!ModelState.IsValid)
+				{
+					ResetOnValidationError();
+					ResetOnValidationError();
+					var data = PopulateNtiFromRequest();
+					var isChecked = data.NtiReasonsForConsidering.Where(c => c.Id != 0);
+					NTIReasonsToConsiderForUI = GetReasonsForUI(NtiModel).ToList();
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Edit.cshtml.cs:81]: This function selects the selected item, so we don't need the logic below it on edit

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs [hunk 2]
```diff
@@ -70,19 +73,28 @@ public async Task<IActionResult> OnPostAsync()
 		{
 			try
 			{
+				var status = Request.Form["status"];
+				var success = int.TryParse(status, out var result);
+				if (!success)
+				{
+					ModelState.AddModelError("options", "Please select a reason for closing NTI under consideration");
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs:80]: You can use the required flag on the radio item to make a radio list mandatory

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs [hunk 3]
```diff
@@ -21,12 +21,16 @@ public class ClosePageModel : AbstractPageModel
 		private readonly INtiUnderConsiderationStatusesCachedService _ntiStatusesCachedService;
 		private readonly ILogger<ClosePageModel> _logger;
 
-		public int NotesMaxLength => 2000;
+		public  int NotesMaxLength => 2000;
 		public IEnumerable<RadioItem> NTIStatuses;
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs:25]: Convert this to use the component then you get all the error formatting for free

**Review by mikestock-nimble — CHANGES_REQUESTED** [review]

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Add.cshtml.cs [hunk 4]
```diff
@@ -43,7 +50,8 @@ public IActionResult OnGet()
 
 			try
 			{
-				NTIReasonsToConsider = GetReasons();
+				LoadPageComponents();
+				NTIReasonsToConsider = GetReasons().ToList();
 				ExtractCaseUrnFromRoute();
```
[the remark this answers is not in the document]
Comment (nimble-dfe) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Add.cshtml.cs:55]: Updated :-)

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Edit.cshtml.cs (hunk 1 above)
Comment (nimble-dfe) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Edit.cshtml.cs:81]: removed dup

Comment (nimble-dfe) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Edit.cshtml.cs:81]: Passed in data isn't a radio button

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs (hunk 3 above)
Comment (nimble-dfe) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs:25]: Changed

File: ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs (hunk 2 above)
Comment (nimble-dfe) [reply ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/NtiUnderConsideration/Close.cshtml.cs:80]: I'm leaving it for now

**Review by mikestock-nimble — APPROVED** [review]

**Pull request merged**